NotifyMe/views/views.py

```diff
@@ -0,0 +1,324 @@
+import logging
+from django.core.exceptions import PermissionDenied
+from django.db import IntegrityError
+from django.core.exceptions import ValidationError
+from rest_framework.response import Response
+from ..serializers import UserSerializer, SubscriptionSerializer, SubscriptionPlanSerializer
+from NotifyMe.models.user import User
+from NotifyMe.models.subscription import Subscription
+from NotifyMe.models.subscriptionPlan import SubscriptionPlan
+from rest_framework import status 
+from rest_framework.views import APIView
+from NotifyMe.services.service import UserService, SubscriptionService, SubscriptionPlanService
+
+
+
+logger = logging.getLogger(__name__)
+
+
+# ---------USER-API---------------- 
+# DONOT CALL MODEL INSIDE VIEW---> ONLY CALL SERVICE AND SERIALIZER
+
+def get_response_data(success, message=None, data=None):
+    response_data = {'success': success}
+    if message:
+        response_data['message'] = message
+    if data is not None:
+        response_data['data'] = data
+    return response_data
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+
+     
+class UserAPI(APIView):
+        
+    def get(self, request):
+        user_service = UserService()
+        try:
+            logger.info("Fetching user data")
+            objects = user_service.get_all_users()
+            serializer = UserSerializer(objects, many=True)
+            logger.info("Successfully fetched user data")
+            return Response(get_response_data(True, data=serializer.data))
+        except PermissionDenied:
+            logger.warning("Permission denied while fetching user data")
+            return Response(get_response_data(False, "You don't have permission to access this data"), status=status.HTTP_403_FORBIDDEN)
+        except Exception as e:
+            logger.error(f"Unexpected error occurred while fetching user data: {e}")
+            return Response(get_response_data(False, "An unexpected error occurred"), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    
+    def post(self, request):
+        try:
+            logger.info("Creating a new user")
+            data = request.data.copy()
```

Comment:
why are copying the data object

Reply:
I am copying data object because while doing create_user method, i will pop the subscription_plan and store it then I will use the subscription_plan object to get the end_time and  to create subscriprion object that will get stored in subscription dataabse when user gets created